jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/cpu/ContextSwitchRateHandler.java

```diff
@@ -0,0 +1,44 @@
+package io.opentelemetry.contrib.jfr.metrics.cpu;
+
+import io.opentelemetry.api.metrics.*;
+import io.opentelemetry.contrib.jfr.Constants;
+import io.opentelemetry.contrib.jfr.metrics.RecordedEventHandler;
+import jdk.jfr.consumer.RecordedEvent;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+public class ContextSwitchRateHandler implements RecordedEventHandler {
+  public static final String EVENT_NAME = "jdk.ThreadContextSwitchRate";
+
+  private final Meter otelMeter;
+  private volatile double value = 0;
+
+  public ContextSwitchRateHandler(Meter otelMeter) {
+    this.otelMeter = otelMeter;
+  }
+
+  public ContextSwitchRateHandler init() {
+    otelMeter.upDownCounterBuilder("jfr.ThreadContextSwitchRate")
+                                     .ofDoubles()
+                                     .setUnit(Constants.PERCENTAGE)
+                                     .buildWithCallback(codm -> codm.observe(value));
+    return this;
+  }
+
+  @Override
+  public void accept(RecordedEvent ev) {
+    value = ev.getDouble("switchRate");
```

Comment:
Instead of building async instruments, you should be able to just create an UpDownCounter (or Histogram) instrument and directly call "record" or "add" here.

Given this is really the full value, UpDownCounter doesn't make much sense, but perhaps Histogram would.